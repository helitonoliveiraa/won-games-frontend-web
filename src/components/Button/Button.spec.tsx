import { FiShoppingCart } from 'react-icons/fi';

import { screen } from '@testing-library/react';

import { renderWithTheme } from '@/utils/tests/helpers';

import { Button } from './index';

describe('<Button/>', () => {
  it('should render the medium size by default', () => {
    const { container } = renderWithTheme(<Button>buy now</Button>);

    expect(screen.getByRole('button', { name: /buy now/i })).toHaveStyle({
      height: '4rem',
      padding: '0.8rem 3.2rem',
      'font-size': '1.4rem',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render small size', () => {
    renderWithTheme(<Button size="small">buy now</Button>);

    expect(screen.getByRole('button', { name: /buy now/i })).toHaveStyle({
      height: '3rem',
      'font-size': '1.2rem',
    });
  });

  it('should render large size', () => {
    renderWithTheme(<Button size="large">buy now</Button>);

    expect(screen.getByRole('button', { name: /buy now/i })).toHaveStyle({
      height: '5rem',
      'font-size': '1.6rem',
      padding: '0.8rem 4.8rem',
    });
  });

  it('should render fullWidth version', () => {
    renderWithTheme(<Button fullWidth>buy now</Button>);

    expect(screen.getByRole('button', { name: /buy now/i })).toHaveStyle({
      width: '100%',
    });
  });

  it('should render icon version', () => {
    renderWithTheme(
      <Button icon={<FiShoppingCart data-testid="icon" />}>buy now</Button>,
    );

    expect(
      screen.getByRole('button', { name: /buy now/i }),
    ).toBeInTheDocument();
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });
});
