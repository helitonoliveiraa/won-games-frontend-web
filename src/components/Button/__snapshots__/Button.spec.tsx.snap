// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render the medium size by default 1`] = `
.c0 {
  border: none;
  border-radius: 0.4rem;
  background: linear-gradient(180deg,#FF5F5F 0%,#f062c0) 50%;
  color: #FAFAFA;
  padding: 0.8rem;
  cursor: pointer;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

<button
  class="c0"
>
  <span>
    buy now
  </span>
</button>
`;
